lxd/certificates.go

```diff
@@ -718,19 +718,58 @@ func doCertificateUpdate(d *Daemon, dbInfo db.Certificate, req api.CertificatePu
 		}
 
 		// Convert to the database type.
-		cert := db.Certificate{
-			// Read-only fields.
+		dbCert := db.Certificate{
 			Certificate: dbInfo.Certificate,
 			Fingerprint: dbInfo.Fingerprint,
+			Restricted:  req.Restricted,
+			Projects:    req.Projects,
+			Name:        req.Name,
+			Type:        reqDBType,
+		}
+
+		if req.Certificate != "" && dbInfo.Certificate != req.Certificate {
+			// Only allow certificate updates by its owner
+			if r.TLS != nil {
```

Comment:
Not quite, we should also allows admins to do it. `rbac.UserIsAdmin(r)` should work for that.